autosubmit/job/job.py

```diff
@@ -1264,49 +1283,61 @@ def retrieve_logfiles(self, platform: Any, raise_error: bool = False) -> Dict[st
+
+        :return int: The total wallclock time in seconds.
+        """
+        total = int(wallclock.total_seconds() * 1.30)
+        total_platform = self._max_possible_wallclock()
+        if not total_platform:
+            total_platform = total
+        if total > total_platform:
+            Log.warning(f"Job {self.name} has a wallclock time '{total} seconds' higher than the maximum allowed by the platform '{total_platform} seconds' "
+                        f"Setting wallclock time to the maximum allowed by the platform.")
+            total = total_platform
+        wallclock_delta = datetime.timedelta(seconds=total)
+        return int(wallclock_delta.total_seconds())
+
+    @staticmethod
+    def parse_time(wallclock):
         regex = re.compile(r'(((?P<hours>\d+):)((?P<minutes>\d+)))(:(?P<seconds>\d+))?')
         parts = regex.match(wallclock)
         if not parts:
-            return
+            return None
         parts = parts.groupdict()
-        if int(parts['hours']) > 0 :
-            format_ = "hour"
-        else:
-            format_ = "minute"
         time_params = {}
         for name, param in parts.items():
             if param:
                 time_params[name] = int(param)
-        return datetime.timedelta(**time_params),format_
+        return datetime.timedelta(**time_params)
 
-    # Duplicated for wrappers and jobs to fix in 4.0.0
-    def is_over_wallclock(self, start_time, wallclock):
+    # TODO Duplicated for wrappers and jobs to fix in 4.1.X but in wrappers is called _is_over_wallclock for unknown reasons
+    def is_over_wallclock(self):
         """
         Check if the job is over the wallclock time, it is an alternative method to avoid platform issues
-        :param start_time:
-        :param wallclock:
         :return:
         """
-        elapsed = datetime.datetime.now() - start_time
-        wallclock,time_format = self.parse_time(wallclock)
-        if time_format == "hour":
-            total = wallclock.days * 24 + wallclock.seconds / 60 / 60
-        else:
-            total = wallclock.days * 24 + wallclock.seconds / 60
-        total = total * 1.30 # in this case we only want to avoid slurm issues so the time is increased by 50%
-        if time_format == "hour":
-            hour = int(total)
-            minute = int((total - int(total)) * 60.0)
-            second = int(((total - int(total)) * 60 -
-                          int((total - int(total)) * 60.0)) * 60.0)
-            wallclock_delta = datetime.timedelta(hours=hour, minutes=minute,
-                                                 seconds=second)
-        else:
-            minute = int(total)
-            second = int((total - int(total)) * 60.0)
-            wallclock_delta = datetime.timedelta(minutes=minute, seconds=second)
-        if elapsed > wallclock_delta:
+        elapsed = datetime.datetime.now() - self.start_time
+        if int(elapsed.total_seconds()) > self.wallclock_in_seconds:
```

Comment:
Not sure if this works as expected?

```py
>>> import datetime
>>> 
>>> start_time = datetime.datetime.now()
>>> 
>>> elapsed = datetime.datetime.now() - start_time
>>> 
>>> print(elapsed)
0:00:11.001284
>>> type(elapsed)
<class 'datetime.timedelta'>
>>> int(elapsed)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'datetime.timedelta'
```

I think it would result in a `TypeError` here?

Reply:
erm, not sure why it is not working in the console

The start_time value didn't change, before it was sending the job.start_time and now it directly calls to self.start_time but the value is the same as before

start_time == self.start_time == time.time()

Reply:
nvm, 

the start_time that is being recieved is 

# paramiko_platform.py line 647-651

```python
            elif job_status in self.job_status['RUNNING']:
                job_status = Status.RUNNING
                if not is_wrapper:
                    if job.status != Status.RUNNING:
                        job.start_time = datetime.datetime.now() # URi: start time
```

Reply:
I believe that

job.py line 2083 - 2084
```python
        if hasattr(self, "start_time"):
            self.start_time = time.time()
```
should be

```python
        if not hasattr(self, "start_time"):
            self.start_time = datetime.datetime.now()
```

Reply:
changed